### PR TITLE
AVO-631: Integrate historical links and share dropdown

### PR DESCRIPTION
**Pull request by cadeban**
## Description
[AVO-631](https://linear.app/electricitymaps/issue/AVO-631/integrate-sharing-drop-down-with-historical-links)
Share urls now contain time aggregates and datetimes. Removed isHourly check now that time aggregate view state is stored in url.

To test, ensure that historical linking and more options dropdown feature flags are on.

Note that [live vs historical share](https://linear.app/electricitymaps/issue/AVO-632/share-live-vs-share-historical-data) is coming in a follow up PR. 

### Double check

- [ ] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [ ] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: web/src/hooks/useGetCurrentUrl.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  const {
+    urlTimeAverage,
+    urlDatetime,
+    zoneId: zoneIdParameter,
+  } = useParams<RouteParameters>();
+  const zId = zoneId || zoneIdParameter;
+  const url =
+    baseUrl +
+    (zId
+      ? `/zone/${zId}/${urlTimeAverage}/${urlDatetime || new Date().toISOString()}`
```
Comment (tonypls) [inline web/src/hooks/useGetCurrentUrl.ts:15]: This isn't floored to 00:00Z, when the URL is not present. We could try it with something like this:

`  const selectedDatetime = useAtomValue(selectedDatetimeIndexAtom);
  const targetDate = urlDatetime ?? new Date(selectedDatetime.datetime).toISOString();`
  
  and would be nice to keep zoneId variable naming over zId

**Review by tonypls — COMMENTED** [review]
Couple of comments!

File: web/src/hooks/useGetCurrentUrl.ts (hunk 1 above)
Comment (cadeban) [reply web/src/hooks/useGetCurrentUrl.ts]: Updated, LMK if I need any additional changes!

**Comment by cadeban** [comment]
@tonypls @VIKTORVAV99 would you please take another pass on this PR when y'all have time? Thanks!

**Pull request merged**